sound/soc/intel/boards/sof_sdw.c

```diff
@@ -797,16 +802,16 @@ static int create_sdw_dailink(struct device *dev, int *be_index,
 		char *name, *cpu_name;
 		int playback, capture;
 		static const char * const sdw_stream_name[] = {
-			"SDW%d-Playback",
-			"SDW%d-Capture",
+			"SDW%d-Playback-%s",
+			"SDW%d-Capture-%s",
```

Comment:
you need to adapt topology to this change ? And how to generate topology name without know the info of sdw codec ? For original design, we don't need to know the detail of sdw codec

Reply:
please focus on real case. As you know, do we consider such case for HDA in machine driver ? No, no customer adopt such design.

Reply:
never say never @RanderWang 
I am sure that some day a creative hardware designer will mix HDAudio and SoundWire, just because they can.

Reply:
@RanderWang This is not for HDA, but for SDW. We will need to use different dai links on a sdca codec if we want to use different functions on it. For example, we need a dai link for headset and the other dai link for DMIC.

@plbossart Fixed. The dai name postfix will only be applied when we specify a dai name in `snd_soc_acpi_link_adr`

Reply:
Do you mean two BE dai in one sdw link ? If so we need to change (1) get BE number function (2) generate BE dai link based on this number

Reply:
> @RanderWang This is not for HDA, but for SDW. We will need to use different dai links on a sdca codec if we want to use different functions on it. For example, we need a dai link for headset and the other dai link for DMIC.
one for playback one for capture , is there any problem for current state ?

Reply:
> > @RanderWang This is not for HDA, but for SDW. We will need to use different dai links on a sdca codec if we want to use different functions on it. For example, we need a dai link for headset and the other dai link for DMIC.
> 
> one for playback one for capture , is there any problem for current state ?

Currently, no. But we will have one for playback and **two** for capture soon. That will be a problem.